torchdata/dataloader2/dataloader2.py

```diff
@@ -97,7 +111,7 @@ def __init__(
             self.datapipe_adapter_fns = datapipe_adapter_fn
         else:
             self.datapipe_adapter_fns = [datapipe_adapter_fn]
-        self.reading_service = reading_service
+        self.reading_service = deepcopy(reading_service)
```

Comment:
Let's try to import your PR to internal to validate if this is working properly with internal RS.

Reply:
+1 on checking existing implementations.